caper/caper_backend.py

```diff
@@ -171,13 +172,94 @@ def __init__(self, db_type=None, db_timeout=None,
             db['password'] = postgresql_password
 
         else:
-            raise Exception('Unsupported DB type {}'.format(db_type))
+            raise ValueError('Unsupported DB type {}'.format(db_type))
 
         if db_timeout is not None and 'db' in self['database']:
             self['database']['db']['connectionTimeout'] = db_timeout
 
 
-class CaperBackendGCP(dict):
+class CaperBackendBase(UserDict):
+    """Base skeleton backend for all backends
+    """
+    CONCURRENT_JOB_LIMIT = None
+    TEMPLATE = {
+        "backend": {
+            "providers": {
+            }
+        }
+    }
+    TEMPLATE_BACKEND = {
+        "actor-factory": None,
+        "config": {
+            "default-runtime-attributes": {
+            },
+            "concurrent-job-limit": None
+        }
+    }
+    def __init__(self, backend_name, d=None):
+        """
+        Args:
+            backend_name: backend name
+            d: dict to override this class on
```

Comment:
I appreciate the fact that it may require quite a bit of pondering (kinda like this: https://devhumor.com/content/uploads/images/May2018/meaningful_variable_name.jpg) , but this variable needs a real name.

Reply:
I see. I changed those parameters' positions (`backend_name` <-> `d`). So that these dict-based classes truly work as a `dict`, which takes the first pos arg as a `dict` to override on `self`. So now we don't need to name this `d` variable.